### ISSUE TITLE
Issue: Script Hanging Due to Missing Timeout Handlers

### ISSUE THREAD

**Issue by 2ndSky95** [Tysonpower/starlinkstatus#53]
## Problem Description
The script can hang indefinitely due to several critical points where network operations lack proper timeout handling:

1. Main hanging points in `pingservers()`:
```bash
# No timeout for server list retrieval:
SERVERS=($(curl -s ${SERVERSURL}))

# No timeout for parallel ping execution:
results=$(printf "%s\n" "${SERVERS[@]}" | parallel --no-notice ping_server {})
```

2. In `collect_lr()`:
```bash
# Speedtest can hang indefinitely:
st=$(speedtest --accept-license --accept-gdpr -f json)
```

3. Main loop issue:
```bash
while true; do
    current_time=$(date +%s)
    # If collect_hr or collect_lr hang, the loop continues but operations are frozen
    if (( current_time % 15 == 0 )); then
        collect_hr        # <- Can hang
    fi
    if (( current_time % lr_interval == 0 )); then
        collect_lr        # <- Can hang
    fi
    sleep 1
done
```

## Suggested Improvements

1. Add user feedback for operations:
```bash
pingservers() {
    echo "Starting ping tests..."
    if ! SERVERS=($(curl -s --max-time 10 ${SERVERSURL})); then
        echo "Error: Could not retrieve server list"
        return 1
    fi
    echo "Server list successfully retrieved"
    # Rest of the code...
}
```

2. Implement basic timeout mechanisms:
```bash
collect_lr() {
    echo "Starting low-res data collection..."
    if ! pingservers; then
        echo "Ping tests failed - skipping"
    fi
    
    if [ $speedtest == true ]; then
        echo "Starting speedtest..."
        if ! st=$(timeout 60 speedtest --accept-license --accept-gdpr -f json); then
            echo "Speedtest failed or timed out after 60 seconds"
            st="{}"
        fi
    fi
    # Rest of the code...
}
```

3. Add status logging:
```bash
# At the start of main loop:
while true; do
    echo "Status check: $(date)"
    current_time=$(date +%s)
    # ... rest of the code
done
```

## User Guidance
If the script hangs, users can:
1. Check log files for error messages
2. Terminate the script using `kill` with the PID from the PID file
3. Restart the script
4. If it happens frequently, consider using the `-o` option for one-time execution and use cron for regular scheduling instead

## Technical Background
The hanging occurs because network operations (curl, ping, speedtest) have no timeout mechanisms implemented. Even after network connectivity is restored, the hanging processes continue to wait indefinitely for a response from their last failed operation.

## Impact
- Script becomes unresponsive
- Resources remain allocated to hanging processes
- Monitoring data collection is interrupted
- Manual intervention required to restore operation

This is a reliability issue that affects long-term monitoring capabilities of the script.

Would you like me to propose a complete solution with full timeout implementations for all operations?

**Comment by Tysonpower** [comment]
Thank you for the suggestions, will work on the improvements :)

**Comment by Tysonpower** [comment]
Should be fixed, released Version 1.31
